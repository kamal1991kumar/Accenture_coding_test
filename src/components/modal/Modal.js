import React from "react";
import PropTypes from "prop-types";

export const Modal = ({ children, onClose, onSave, isSaveDisabled }) => (
  <div className="modal">
    <div className="modal__content">
      <h4 className="modal__heading" data-testid="heading">
        Dialog box
      </h4>
      <div className="modal__body" data-testid="children">
        {children}
      </div>
      <div className="modal__footer">
        <button
          className="modal__btn modal__btn-cancel"
          data-testid="close"
          onClick={onClose}
        >
          Cancel
        </button>
        <button
          className="modal__btn modal__btn-save"
          disabled={isSaveDisabled}
          onClick={onSave}
          data-testid="save"
        >
          Save
        </button>
      </div>
    </div>
  </div>
);

Modal.propTypes = {
  children: PropTypes.oneOfType([PropTypes.array, PropTypes.string]),
  onClose: PropTypes.func.isRequired,
  onSave: PropTypes.func.isRequired,
  isSaveDisabled: PropTypes.bool.isRequired,
};
