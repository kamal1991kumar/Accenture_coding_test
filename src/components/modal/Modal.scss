.modal {
  position: absolute;
  background: rgba(0,0,0,.3);
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 6px;

  &__content {
    width: 100%;
    background: #fff;
    border-radius: 5px;
    box-shadow: 0 0 5px rgba(0,0,0,.5);
  }
  &__heading, &__body, &__footer{
    padding: 10px;
  }
  &__heading {
    font-weight: normal;
    font-size: 12px;
    border-bottom: solid 1px #ebebeb;
  }
  &__footer {
    padding-top: 0;
    text-align: right;
  }
  &__btn {
    font-size: 10px;
    margin-left: 5px;
    cursor: pointer;
    border:solid 1px #7f7f7f;
    border-radius: 3px;
    padding: 5px 8px;
    &-cancel {
      background: #fff;
      color: #7f7f7f;
    }
    &-save {
      background: #7f7f7f;
      color: #fff;
    }
    &:disabled {
      cursor: not-allowed;
      opacity: .6;
    }
  }
}