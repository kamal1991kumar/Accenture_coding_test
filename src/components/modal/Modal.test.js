import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Modal from "./index";

const props = {
  isSaveDisabled: false,
  onClose: jest.fn(),
  onSave: jest.fn(),
};

const TestModal = (p) => (
  <Modal {...p} {...props}>
    {p.children}
  </Modal>
);

describe("Modal Component", () => {
  test("Should render on screen", () => {
    const { container } = render(<TestModal />);
    expect(container).toMatchSnapshot();
  });

  test("Should render heading", () => {
    const { getByTestId } = render(<TestModal />);
    expect(getByTestId("heading").textContent).toBe("Dialog box");
  });

  test("Should render body", () => {
    const { getByTestId } = render(<TestModal>hello world</TestModal>);
    expect(getByTestId("children").textContent).toBe("hello world");
  });

  test("Should call close function click on cancel button", () => {
    const { getByTestId } = render(<TestModal />);
    const btn = getByTestId("close");
    fireEvent.click(btn);
    expect(btn.textContent).toBe("Cancel");
    expect(props.onClose.mock.calls.length).toEqual(1);
  });

  test("Should call save function click on save button", () => {
    const { getByTestId } = render(<TestModal />);
    const btn = getByTestId("save");
    fireEvent.click(btn);
    expect(btn.textContent).toBe("Save");
    expect(btn.disabled).toBe(false);
    expect(props.onSave.mock.calls.length).toEqual(1);
  });
});
