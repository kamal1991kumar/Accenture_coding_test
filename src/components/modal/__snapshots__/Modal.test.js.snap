// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component Should render on screen 1`] = `
<div>
  <div
    class="modal"
  >
    <div
      class="modal__content"
    >
      <h4
        class="modal__heading"
        data-testid="heading"
      >
        Dialog box
      </h4>
      <div
        class="modal__body"
        data-testid="children"
      />
      <div
        class="modal__footer"
      >
        <button
          class="modal__btn modal__btn-cancel"
          data-testid="close"
        >
          Cancel
        </button>
        <button
          class="modal__btn modal__btn-save"
          data-testid="save"
        >
          Save
        </button>
      </div>
    </div>
  </div>
</div>
`;
